src/Solnet.Extensions/TokenWallet.cs

```diff
@@ -181,6 +193,106 @@ private TokenWallet(ITokenWalletRpcProxy client, ITokenMintResolver mintResolver
+        /// <param name="publicKey">The account public key.</param>
+        /// <param name="commitment">The state commitment to consider when querying the ledger state.</param>
+        /// <returns>A TokenWallet task that will trigger once the batch has executed.</returns>
+        public static Task<TokenWallet> LoadAsync(SolanaRpcBatchWithCallbacks batch,
+                                                  ITokenMintResolver mintResolver,
+                                                  string publicKey,
+                                                  Commitment commitment = Commitment.Finalized)
+        {
+            if (batch == null) throw new ArgumentNullException(nameof(batch));
+            if (mintResolver == null) throw new ArgumentNullException(nameof(mintResolver));
+            if (publicKey == null) throw new ArgumentNullException(nameof(publicKey));
+
+            // create the task source
+            var taskSource = new TaskCompletionSource<TokenWallet>();
+            var success = 0;
+            var fail = 0;
+            ulong lamports = 0;
+            List<TokenAccount> tokenAccounts = null;
+
+            // function to create a token wallet when both callbacks have responsed (in any order)
+            Action<Exception> wrapUp = ex =>
+            {
+                if (success == 2)
+                {
+                    var tokenWallet = new TokenWallet(mintResolver, new PublicKey(publicKey));
+                    tokenWallet.Lamports = lamports;
+                    tokenWallet._tokenAccounts = tokenAccounts;
+                    taskSource.SetResult(tokenWallet);
+                }
+                else if (fail + success == 2)
+                    taskSource.SetException(new ApplicationException("Failed to load TokenWallet via Batch"));
+            };
+
+            // get sol balance 
+            batch.GetBalance(publicKey, commitment, callback: (x, ex) =>
+            {
+                // handle balance response
+                lock (taskSource)
+                {
+                    if (x != null)
+                    {
+                        lamports = x.Value;
+                        success += 1;
+                    }
+                    else
+                        fail += 1;
+                }
+
+                // finished?
+                wrapUp.Invoke(ex);
+            });
+
+            // load token accounts
+            batch.GetTokenAccountsByOwner(publicKey, null, TokenProgram.ProgramIdKey, commitment, callback: (x, ex) =>
+            {
+                // handle token list response
+                lock (taskSource)
+                {
+                    if (x != null)
+                    {
+                        tokenAccounts = x.Value;
+                        success += 1;
+                    }
+                    else
+                        fail += 1;
+                }
+
+                // finished?
+                wrapUp.Invoke(ex);
+            });
+
```

Comment:
My pea brain was confused by the flow here, but I assume the logic is to be able to batch more than just these two requests without forcing a flush?